eppoclient/client.go

```diff
@@ -80,6 +81,76 @@ func (ec *EppoClient) GetJSONAssignment(flagKey string, subjectKey string, subje
 	return variation, err
 }
 
+type BanditResult struct {
+	Variation string
+	Action    *string
+}
+
+func (ec *EppoClient) GetBanditAction(flagKey string, subjectKey string, subjectAttributes ContextAttributes, actions map[string]ContextAttributes, defaultVariation string) BanditResult {
+	// ignoring the error here as we can always proceed with default variation
+	variation, _ := ec.GetStringAssignment(flagKey, subjectKey, subjectAttributes.toGenericAttributes(), defaultVariation)
+
+	bandit, err := ec.configurationStore.getBanditConfiguration(variation)
+	if err != nil {
+		// no bandit configuration
+		return BanditResult{
+			Variation: variation,
+			Action:    nil,
+		}
+	}
+
+	evaluation := bandit.ModelData.evaluate(banditEvaluationContext{
+		flagKey:           flagKey,
+		subjectKey:        subjectKey,
+		subjectAttributes: subjectAttributes,
+		actions:           actions,
+	})
+	if evaluation.result == nil {
+		return BanditResult{
+			Variation: variation,
+			Action:    nil,
+		}
+	}
+
+	if logger, ok := ec.logger.(BanditActionLogger); ok {
```

Comment:
One difference from Python SDK is that empty evaluations are not logged. Empty evaluations can only happen if `actions` passed is empty, so I'm not sure how helpful that is.

It's a bit cumbersome to work with nullable values in Go, so I decided to not log them. I hope that's fine. But if not—let me know and I'll rework it.

cc @schmit

Reply:
Across the board, for bandits and non-bandit flags, if our evaluation hit an error and we're going to return the default, we do not want to log as is its not an "intentional" assignment.

I think if empty actions are passed, we'd want to return (and not log) the default value.

Reply:
Moved this discussion to Slack here: https://eppo-group.slack.com/archives/C05R6BK5BB9/p1718906479473079